### PR TITLE
feat: enable search with multiple types and statuses (MAPCO-4427, MAPCO-4430)

### PR DESCRIPTION
**Pull request by RonitKissis**
added new POST route to enable an array of jobTypes and Status types when querying Job table

| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                   |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                      |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: src/common/dataModels/jobs.ts [hunk 1]
```diff
@@ -26,6 +26,21 @@ export interface IFindJobsRequest {
   domain?: string;
 }
 
+export interface IFindJobsByCriteriaBody {
+  resourceId?: string;
+  version?: string;
+  isCleaned?: boolean;
+  status?: [OperationStatus];
```
Comment (CL-SHLOMIKONCHA) [inline src/common/dataModels/jobs.ts]:
```suggestion
  status?: OperationStatus[];
```

File: src/common/dataModels/jobs.ts [hunk 2]
```diff
@@ -26,6 +26,21 @@ export interface IFindJobsRequest {
...
+  type?: [string];
```
Comment (CL-SHLOMIKONCHA) [inline src/common/dataModels/jobs.ts]:
```suggestion
  type?: string[];
```

File: tests/integration/jobs/jobs.spec.ts [hunk 3]
```diff
@@ -589,6 +589,67 @@ describe('job', function () {
...
+        expect(where).toHaveBeenCalledTimes(1);
+        expect(andWhere).toHaveBeenCalledTimes(2);
+        expect(getMany).toHaveBeenCalledTimes(1);
+
+        const jobs = response.body as unknown;
+        expect(jobs).toEqual([jobModel]);
+        expect(response).toSatisfyApiSpec();
+      });
+
+      it('should no jobs and return 200', async function () {
```
Comment (CL-SHLOMIKONCHA) [inline tests/integration/jobs/jobs.spec.ts]: change test name to: 'should not find matched jobs and return status 200 with an empty array'

File: tests/integration/jobs/jobs.spec.ts [hunk 4]
```diff
@@ -589,6 +589,67 @@ describe('job', function () {
...
+        const select = jobRepositoryMocks.queryBuilder.select;
+        const from = jobRepositoryMocks.queryBuilder.from;
+        const where = jobRepositoryMocks.queryBuilder.where;
+        const andWhere = jobRepositoryMocks.queryBuilder.andWhere;
+        const getMany = jobRepositoryMocks.queryBuilder.getMany;
+        getMany.mockResolvedValue([]);
+
+        const response = await requestSender.findJobs(filter);
+
+        expect(response.status).toBe(httpStatusCodes.OK);
```
Comment (CL-SHLOMIKONCHA) [inline tests/integration/jobs/jobs.spec.ts:642]: add expect(response.body) to be empty array

File: tests/integration/jobs/jobs.spec.ts (hunk 3 above)
Comment (RonitKissis) [reply tests/integration/jobs/jobs.spec.ts]: done

File: tests/integration/jobs/jobs.spec.ts (hunk 4 above)
Comment (RonitKissis) [reply tests/integration/jobs/jobs.spec.ts:642]: done

File: src/common/dataModels/jobs.ts (hunk 2 above)
Comment (RonitKissis) [reply src/common/dataModels/jobs.ts]: done

File: src/common/dataModels/jobs.ts (hunk 1 above)
Comment (RonitKissis) [reply src/common/dataModels/jobs.ts]: done

**Review by CL-SHLOMIKONCHA — APPROVED** [review]

File: src/common/dataModels/jobs.ts [hunk 5]
```diff
@@ -26,6 +26,21 @@ export interface IFindJobsRequest {
   domain?: string;
 }
 
+export interface IFindJobsByCriteriaBody {
+  resourceId?: string;
+  version?: string;
+  isCleaned?: boolean;
+  status?: OperationStatus[];
```
Comment (CL-SHLOMIKONCHA) [inline src/common/dataModels/jobs.ts]: statuses?

File: src/common/dataModels/jobs.ts [hunk 6]
```diff
@@ -26,6 +26,21 @@ export interface IFindJobsRequest {
...
+  type?: string[];
```
Comment (CL-SHLOMIKONCHA) [inline src/common/dataModels/jobs.ts]: types?

File: openapi3.yaml [hunk 7]
```diff
@@ -668,6 +689,43 @@ components:
...
+          type: string
+        version:
+          type: string
+        isCleaned:
+          type: boolean
+        status:
+          type: array
+          items:
+            $ref: '#/components/schemas/status'
+        type:
```
Comment (CL-SHLOMIKONCHA) [inline openapi3.yaml]: types?

File: openapi3.yaml [hunk 8]
```diff
@@ -668,6 +689,43 @@ components:
...
+      type: object
+      description: get jobs criteria
+      properties:
+        resourceId:
+          type: string
+        version:
+          type: string
+        isCleaned:
+          type: boolean
+        status:
```
Comment (CL-SHLOMIKONCHA) [inline openapi3.yaml]: statuses?

File: tests/integration/jobs/jobs.spec.ts [hunk 9]
```diff
@@ -589,6 +589,68 @@ describe('job', function () {
...
+        const jobs = response.body as unknown;
+        expect(jobs).toEqual([jobModel]);
+        expect(response).toSatisfyApiSpec();
+      });
+
+      it('should not find matched jobs and return status 200 with an empty array', async function () {
+        const filter = {
+          isCleaned: true,
+          resourceId: '1',
+          status: ['Pending'],
```
Comment (CL-SHLOMIKONCHA) [inline tests/integration/jobs/jobs.spec.ts]: statuses

File: tests/integration/jobs/jobs.spec.ts [hunk 10]
```diff
@@ -589,6 +589,68 @@ describe('job', function () {
...
+          type: ['2'],
```
Comment (CL-SHLOMIKONCHA) [inline tests/integration/jobs/jobs.spec.ts]: types

File: src/DAL/repositories/jobRepository.ts [hunk 11]
```diff
@@ -69,6 +70,44 @@ export class JobRepository extends GeneralRepository<JobEntity> {
...
+      }
+    }
+    if (req.shouldReturnTasks !== false) {
+      filter.relations = ['tasks'];
+    }
+    const queryBuilder = this.createQueryBuilder().select('job').from(JobEntity, 'job').where(filter);
+    if ((req.type?.length ?? 0) > 0) {
+      queryBuilder.andWhere('job.type IN (:...types)', { types: req.type });
+    }
+    if ((req.status?.length ?? 0) > 0) {
```
Comment (CL-SHLOMIKONCHA) [inline src/DAL/repositories/jobRepository.ts]: req.statuses?

File: src/DAL/repositories/jobRepository.ts [hunk 12]
```diff
@@ -69,6 +70,44 @@ export class JobRepository extends GeneralRepository<JobEntity> {
...
+    for (const key of Object.keys(filter)) {
+      if (filter[key] == undefined) {
+        delete filter[key];
+      }
+    }
+    if (req.shouldReturnTasks !== false) {
+      filter.relations = ['tasks'];
+    }
+    const queryBuilder = this.createQueryBuilder().select('job').from(JobEntity, 'job').where(filter);
+    if ((req.type?.length ?? 0) > 0) {
```
Comment (CL-SHLOMIKONCHA) [inline src/DAL/repositories/jobRepository.ts]: req.types?

**Review by CL-SHLOMIKONCHA — APPROVED** [review]
Good job!

**Review by netanelC — APPROVED** [review]

**Pull request merged**